ShellScript/test/syntax_test_bash.sh

```diff
@@ -865,6 +865,46 @@ coproc foobar {
 # 3.3 Shell Functions                                              #
 ####################################################################
 
+   ()
+#^^ - meta.function
+#  ^ meta.function.parameters.shell
+#   ^ meta.function.parameters.shell
+#    ^ meta.function.shell
+
+   ()
+   {}
```

Comment:
Funny enough, this is perfectly valid zsh syntax:

```zsh
% () () () { echo hi; }
hi
```

Reply:
I'd expect `() () (echo hi;)` to work as the first `()` would be the function name, the second could be the arguments dummy and the 3rd one the function body.

But honestly, we can find dozens of such inconsistencies in all script/shell syntaxes created in the stoneage of PCs's (80s and 90s) when chaos and adhoc implementations drove development. All of those lack correctly structured syntax which causes them to be filled with dozens of ambigulties.

If we want to highlight them all, syntax will probably grow to several Megabytes.

Reply:
I mean, isn't that awesome? :D